app/vmalert/utils/auth.go

```diff
@@ -0,0 +1,76 @@
+package utils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+// Authorizer is a func which adds Auth header
+// to the given request
+type Authorizer func(r *http.Request)
+
+const authHeader = "Authorization"
+
+// RequestAuthorizer returns Authorizer based on the
+// provided auth params.
+// Returns nil if all auth params are empty.
+func RequestAuthorizer(baUsername, baPassword, bearerToken, bearerTokenFile string) (Authorizer, error) {
```

Comment:
WDYT about reusing code from https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/promauth/config.go#L250

 It caches auth header and allows to dynamically update password from password_file 

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/promauth/config.go#L176

Reply:
Agreed - this would be awesome from maintainability PoV!

Something like the following:

```go
ba := &promauth.BasicAuthConfig{
  Username: baUsername,
  Password: baPassword,
}
ac, err := promauth.NewConfig(".", nil, ba, bearerToken, bearerTokenFile, nil, nil)
if err != nil {
  return nil, err
}
return func(r *http.Request) {
  ah:= ac.GetAuthHeader()
  if ah != "" {
    r.Header.Set("Authorization", ah)
  }
}, nil
```